spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerEndpoint.java

```diff
@@ -202,10 +223,25 @@ public void setTopicPartitions(TopicPartitionOffset... topicPartitions) {
 	/**
 	 * Return the topicPartitions for this endpoint.
 	 * @return the topicPartitions for this endpoint.
+	 * @deprecated in favor of {@link #getTopicPartitionsToAssign()}
+	 */
+	@Deprecated
+	@SuppressWarnings("deprecation")
+	@Override
+	public Collection<org.springframework.kafka.support.TopicPartitionInitialOffset> getTopicPartitions() {
+		return Collections.unmodifiableCollection(this.topicPartitions.stream()
+				.map(TopicPartitionOffset::toTPIO)
```

Comment:
Same "ball of mud" isolation concern: `TopicPartitionInitialOffset.fromTopicPartitionOffset()`.